examples/http3-client.rs

```diff
@@ -71,9 +72,31 @@ fn main() {
     let poll = mio::Poll::new().unwrap();
     let mut events = mio::Events::with_capacity(1024);
 
+    // Take a look at server address resolved to check if it's ipv4 or ipv6.
+    // Depending on the IP family, bind_addr will be default address of
+    // v4 or v6 for calling bind() later.
+    // This workaround is to work with MacOS (or BSD variants which
+    // doesn't allow v4 and v6 can be bind() in one socket).
+    // Note that linux doesn't need this because it can handle v4 and v6 socket
+    // when bind() with "::".
+    let mut addrs_iter = url.to_socket_addrs().unwrap();
+    let addr = addrs_iter.next(); // get next IP address resolved from URL hostname
+    let bind_addr;
+    match addr {
+        Some(ip) => {
+            debug!("addr = {:?}", ip);
+            bind_addr = if ip.is_ipv4() {
+                "0.0.0.0:0" // v4
+            } else {
+                "[::]:0" // v6
+           }
+        },
+        None => panic!("IP address not found"),
+    }
+
     // Create the UDP socket backing the QUIC connection, and register it with
     // the event loop.
-    let socket = std::net::UdpSocket::bind("[::]:0").unwrap();
+    let socket = std::net::UdpSocket::bind(bind_addr).unwrap();
     socket.connect(&url).unwrap();
```

Comment:
We should reuse the `peer_addr` variable from above here, to avoid doing another DNS resolution (best case is that we resolve the domain twice for no reason, worst case is that the second resolution returns a different result).

Reply:
We can use `peer_addr` instead of `&url`.